runtime/server/generate_chart.go

```diff
@@ -0,0 +1,134 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/drivers"
+	"github.com/rilldata/rill/runtime/pkg/activity"
+	"github.com/rilldata/rill/runtime/pkg/observability"
+	"github.com/rilldata/rill/runtime/server/auth"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+func (s *Server) GenerateChartSpec(ctx context.Context, req *runtimev1.GenerateChartSpecRequest) (*runtimev1.GenerateChartSpecResponse, error) {
+	observability.AddRequestAttributes(ctx,
+		attribute.String("args.instance_id", req.InstanceId),
+		attribute.String("args.resolver", req.Resolver),
+		attribute.String("args.resolver_property", marshalResolverProperties(req.ResolverProperties.AsMap())),
+		// attribute.String("args.prompt", req.Prompt), // Adding this might be a privacy issue
```

Comment:
These metrics are not emitted on local, so you can uncomment for now for completeness. If we ever emit these from local, we will anyway need to go over all endpoints and add hashing for sensitive fields.